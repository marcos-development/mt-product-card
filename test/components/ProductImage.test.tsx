import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';
import { ProductImage } from '../../src/components/ProductImage';



describe('Pruebas en ProductImage', () => {
    
    test('Debe de cargarse correctamente', () => {
        
        const wrapper = renderer.create(
            <ProductImage img="http://hola.jpg" className="custom-class" />
        )
        expect( wrapper.toJSON() ).toMatchSnapshot();

    })

    test('Debe comprobar que existe la imagen', () => {

        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )
        expect( wrapper.toJSON() ).toMatchSnapshot();

    })

})
