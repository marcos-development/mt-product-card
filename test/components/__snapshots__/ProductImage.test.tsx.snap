// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage Debe comprobar que existe la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee Mug"
    className="productImg "
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en ProductImage Debe de cargarse correctamente 1`] = `
<img
  alt="Coffee Mug"
  className="productImg custom-class"
  src="http://hola.jpg"
/>
`;
